gitlab/resource_gitlab_repository_file.go

```diff
@@ -0,0 +1,209 @@
+package gitlab
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+const encoding = "base64"
+
+func resourceGitLabRepositoryFile() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGitlabRepositoryFileCreate,
+		Read:   resourceGitlabRepositoryFileRead,
+		Update: resourceGitlabRepositoryFileUpdate,
+		Delete: resourceGitlabRepositoryFileDelete,
+
+		// the schema matches https://docs.gitlab.com/ee/api/repository_files.html#create-new-file-in-repository
+		// However, we don't support the `encoding` parameter as it seems to be broken.
+		// Only a value of `base64` is supported, all others, including the documented default `text`, lead to
+		// a `400 {error: encoding does not have a valid value}` error.
+		Schema: map[string]*schema.Schema{
+			"project": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"file_path": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"branch": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"start_branch": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"author_email": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"author_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"content": {
+				Type:     schema.TypeString,
+				Required: true,
```

Comment:
Maybe this should have a validate func for base64?

Reply:
implemented!